src/components/application_manager/test/hmi_capabilities_test.cc

```diff
@@ -286,84 +283,83 @@ class HMICapabilitiesTest : public ::testing::Test {
+    hmi_capabilities_.reset();
   }
 
-  static void DeleteFileIfExists(const std::string& file_path) {
+  void CreateFile(const std::string& file_name) {
+    file_system::CreateFile(file_name);
+    ASSERT_TRUE(file_system::FileExists(file_name));
+  }
+
+  void DeleteFileIfExists(const std::string& file_path) {
     if (file_system::FileExists(file_path)) {
       EXPECT_TRUE(::file_system::DeleteFile(file_path));
     }
   }
 
-  void SetCooperating();
+  void SetUpLoadCapabilitiesFromFileExpectation() {
+    ON_CALL(mock_app_mngr_, IsSOStructValid(_, _)).WillByDefault(Return(true));
+
+    EXPECT_CALL(*(MockMessageHelper::message_helper_mock()),
+                CommonLanguageFromString(_))
+        .WillRepeatedly(Invoke(TestCommonLanguageFromString));
+    EXPECT_CALL(*(MockMessageHelper::message_helper_mock()),
+                CommonLightNameFromString(_))
+        .WillRepeatedly(Invoke(TestCommonLightNameFromString));
+  }
+
+  std::shared_ptr<HMICapabilitiesImpl> hmi_capabilities_;
   NiceMock<MockApplicationManager> mock_app_mngr_;
-  NiceMock<event_engine_test::MockEventDispatcher> mock_event_dispatcher;
-  resumption::LastStateWrapperPtr last_state_wrapper_;
+  NiceMock<event_engine_test::MockEventDispatcher> mock_event_dispatcher_;
   NiceMock<MockApplicationManagerSettings> mock_application_manager_settings_;
-  std::shared_ptr<HMICapabilitiesForTesting> hmi_capabilities_test;
-  const std::string file_name_;
-  const std::string file_cache_name_;
-  application_manager_test::MockRPCService mock_rpc_service_;
+  resumption::LastStateWrapperPtr last_state_wrapper_;
+  smart_objects::CSmartSchema schema_;
 };
 
-TEST_F(HMICapabilitiesTest, LoadCapabilitiesFromFile) {
-  DeleteFileIfExists(kAppInfoDataFile);
-
-  EXPECT_CALL(*(MockMessageHelper::message_helper_mock()),
-              CommonLanguageFromString(_))
-      .WillRepeatedly(Invoke(TestCommonLanguageFromString));
-  EXPECT_CALL(*(MockMessageHelper::message_helper_mock()),
-              CommonLightNameFromString(_))
-      .WillRepeatedly(Invoke(TestCommonLightNameFromString));
-  ASSERT_TRUE(hmi_capabilities_test->LoadCapabilitiesFromFile());
+TEST_F(HMICapabilitiesTest,
+       Init_CheckActiveLanguages_SuccesSetupDefaultLanguages) {
+  hmi_capabilities_->Init(last_state_wrapper_);
 
-  // Check active languages
   EXPECT_EQ(hmi_apis::Common_Language::EN_US,
-            hmi_capabilities_test->active_ui_language());
+            hmi_capabilities_->active_ui_language());
   EXPECT_EQ(hmi_apis::Common_Language::EN_US,
-            hmi_capabilities_test->active_vr_language());
+            hmi_capabilities_->active_vr_language());
   EXPECT_EQ(hmi_apis::Common_Language::EN_US,
-            hmi_capabilities_test->active_tts_language());
+            hmi_capabilities_->active_tts_language());
+}
 
-  // Check UI languages
+TEST_F(HMICapabilitiesTest,
+       LoadCapabilitiesFromFile_CheckUILanguages_SuccessLoadAndConvert) {
+  SetUpLoadCapabilitiesFromFileExpectation();
```

Comment:
@ydementieiev do we need to use `SetUpLoadCapabilitiesFromFileExpectation` for all test cases?

Reply:
@mvorobio no we don't need , I changed name to more [correct](https://github.com/LuxoftSDL/sdl_core/pull/17/commits/679f3d332f8eab9b627ab268df7009b457cde7b3#diff-c1d0a5558de3ab3e31ce8813484ebefeR326)